packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -128,14 +101,19 @@ export const SelectPopover = styled( Ariakit.SelectPopover )`
 `;
 
 export const SelectItem = styled( Ariakit.SelectItem )`
-	display: flex;
-	align-items: center;
-	justify-content: space-between;
-	padding: ${ ITEM_PADDING };
-	font-size: ${ CONFIG.fontSize };
-	line-height: 2.15rem; // TODO: Remove this in default but keep for back-compat in legacy
-	&[data-active-item] {
-		background-color: ${ COLORS.theme.gray[ 300 ] };
+	// Necessary to override V1 styles passed via the legacy
+	// '.components-custom-select-control__item' class
```

Comment:
> Do you mean [these styles](https://github.com/WordPress/gutenberg/blob/0ea6751e73c4f19d48b51f4f96cc73e7edee56f5/packages/components/src/custom-select-control/style.scss#L44-L75)?

Yup!

> I guess we can remove the && hack right after we replace the v1 with v2 legacy

That's what I had in mind as well — I've added an extra item in the tracking issue so that we don't forget

> we might just want to remove the [v1 stylesheet import](https://github.com/WordPress/gutenberg/blob/0ea6751e73c4f19d48b51f4f96cc73e7edee56f5/packages/components/src/style.scss#L22) while we test v2 legacy styling because there are probably other weird interference as well.

That's a good point. Let's schedule some testing with/without importing V1 styles once all current style fixes are merged — added that to the tracking issues too.

Reply:
Another way I just thought of is to comment out the legacy classnames we're passing in `legacy-component/index.tsx` until we're ready to remove the v1 implementation and styles.

Your call, but that might be simpler than cleaning up the small hacks we're adding in this PR.

Reply:
Clever! I went ahead and followed your suggestion — seems to work well!

When gradually swapping to legacy V2 we'll probably have to enable them (at least while testing) so that overrides like [this one](https://github.com/WordPress/gutenberg/blob/9764ee77d87ec000bf478cba9c5dc60709c97127/packages/block-editor/src/hooks/position.scss#L2-L4) can take effect — unless we ignore those overrides and have the component behave "vanilla".